docs/regression.ipynb

```diff
@@ -0,0 +1,696 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "x, y = prepared_data[:, 0], prepared_data[:, 1]\n",
+    "m, b = coef\n",
+    "line_x = np.linspace(np.min(x), np.max(x), 100)\n",
+    "line_y = m * line_x + b\n",
+    "\n",
+    "plt.scatter(x, y, color='blue', label='Data Points')\n",
+    "plt.plot(line_x, line_y, color='red', label=f'Line: y = {m}x + {b}')\n",
+    "\n",
+    "plt.xlabel('X-axis')\n",
+    "plt.ylabel('Y-axis')\n",
+    "plt.title('Scatter Plot with DP-Theil-Sen Regression')\n",
+    "plt.legend()\n",
+    "\n",
+    "plt.show()"
+   ]
```

Comment:
Naively, something seems to be wrong here. Visually, it seems like there's an obvious correlation. Maybe the data is just very noisy, or you need a larger k? I also wonder if there's a math problem: It seems suspicious that the regression line goes almost through the origin, when year 0 is really arbitrary.